WordPress/src/main/java/org/wordpress/android/ui/me/MyProfileActivity.java

```diff
@@ -0,0 +1,143 @@
+    private WPTextView mAboutMe;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        AccountHelper.getDefaultAccount().fetchAccountSettings(new ApiHelper.GenericCallback() {
+            @Override
+            public void onSuccess() {
+                refreshDetails();
+            }
+
+            @Override
+            public void onFailure(ApiHelper.ErrorType errorType, String errorMessage, Throwable throwable) {
+            }
+        });
+
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setHomeButtonEnabled(true);
+            actionBar.setDisplayHomeAsUpEnabled(true);
+        }
+        setContentView(R.layout.my_profile_activity);
+
+        setTitle(R.string.my_profile);
+
+        mFirstName = (WPTextView) findViewById(R.id.first_name);
+        mLastName = (WPTextView) findViewById(R.id.last_name);
+        mDisplayName = (WPTextView) findViewById(R.id.display_name);
+        mAboutMe = (WPTextView) findViewById(R.id.about_me);
+
+        refreshDetails();
+
+        findViewById(R.id.first_name_row).setOnClickListener(createOnClickListener(getString(org.wordpress.android.R.string.first_name), null, mFirstName));
+        findViewById(R.id.last_name_row).setOnClickListener(createOnClickListener(getString(R.string.last_name), null, mLastName));
+        findViewById(R.id.display_name_row).setOnClickListener(createOnClickListener(getString(R.string.public_display_name), null, mDisplayName));
+        findViewById(R.id.about_me_row).setOnClickListener(createOnClickListener(getString(R.string.about_me), getString(R.string.about_me_hint), mAboutMe));
+    }
+
+    @Override
+    public void finish() {
+        super.finish();
+        ActivityLauncher.slideOutToRight(this);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(final MenuItem item) {
+        if (item.getItemId() == android.R.id.home) {
+            onBackPressed();
+            return true;
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    private void refreshDetails() {
+        Account account = AccountHelper.getDefaultAccount();
+        updateLabel(mFirstName, account != null ? StringUtils.unescapeHTML(account.getFirstName()) : null);
+        updateLabel(mLastName, account != null ? StringUtils.unescapeHTML(account.getLastName()) : null);
+        updateLabel(mDisplayName, account != null ? StringUtils.unescapeHTML(account.getDisplayName()) : null);
+        updateLabel(mAboutMe, account != null ? StringUtils.unescapeHTML(account.getAboutMe()) : null);
+    }
+
+    private void updateMyProfileForLabel(TextView textView) {
+        Map<String, String> params = new HashMap<>();
+        params.put(restParamForTextView(textView), textView.getText().toString());
+        AccountHelper.getDefaultAccount().postAccountSettings(params, new ApiHelper.GenericCallback() {
```

Comment:
I've tested the suggested method with [this diff](https://gist.github.com/oguzkocer/501491fe93a26eae041d). The problem with this approach is, if the user just goes back to Me screen, she will see the old display name and it'll be updated afterwards. Also, for some reason, during my tests, the display name never gets updated. The event is never triggered in `MeFragment` even though it used to be triggered in `MyProfileActivity`.

Since both approaches have some kind of downside, after this discovery, I'd prefer if we just assume the connection is going to be generally fast enough and users won't really do a lot of edits one after the other when the connection is slow. Having said that, I am open to suggestions on improving that diff and giving it another shot.

P.S: I didn't add in `NetworkUtils.isNetworkAvailable()` in the diff. I was going to add that after I tested things out.